compiler/src/typed/parmatch.re

```diff
@@ -1831,8 +1891,13 @@ let untype_constant =
     )
   | Const_number(Const_number_bigint({bigint_rep})) =>
     Parsetree.PConstNumber(Parsetree.PConstNumberInt(bigint_rep))
+  | Const_int8(i) => Parsetree.PConstInt8(Int32.to_string(i))
+  | Const_int16(i) => Parsetree.PConstInt16(Int32.to_string(i))
   | Const_int32(i) => Parsetree.PConstInt32(Int32.to_string(i))
   | Const_int64(i) => Parsetree.PConstInt64(Int64.to_string(i))
+  | Const_uint8(i) => Parsetree.PConstUint8(false, Printf.sprintf("%lu", i))
+  | Const_uint16(i) =>
+    Parsetree.PConstUint16(false, Printf.sprintf("%lu", i))
```

Comment:
`Lu` is the ocaml format specifier for 64-bit unsigned numbers while lowercase `lu` is for 32-bit; since ocaml doesn't have 8 or 16 bit numbers natively I just store short numbers in a regular `int32`